cpp/include/cudf/copying.hpp

```diff
@@ -331,19 +376,155 @@ std::vector<gdf_column*> slice(gdf_column const &          input_column,
  *
  * Example:
  * input:   {10, 12, 14, 16, 18, 20, 22, 24, 26, 28}
- * indices: {2, 5, 9}
+ * splits: {2, 5, 9}
  * output:  {{10, 12}, {14, 16, 18}, {20, 22, 24, 26}, {28}}
  *
  * @param[in] input_column  The input column whose rows will be split.
- * @param[in] indices       An device array of indices that are used to divide the input
- * column into multiple columns.
+ * @param[in] splits        An device array of indices that are used to divide
+ * the input column into multiple columns.
+ * @param[in] num_splits   Number of splits in the splits indices array
  * @return A std::vector of gdf_column*, each of which may have a different size
  * a different number of rows.
  */
-std::vector<gdf_column*> split(gdf_column const &          input_column,
-                               gdf_index_type const*      indices,
-                               gdf_size_type              num_indices);
+std::vector<gdf_column*> split(gdf_column const &         input_column,
+                               gdf_index_type const*      splits,
+                               gdf_size_type              num_splits);
+
+
+/**
+ * @brief Splits a table (including null values) into a set of tables across
+ * rows according to a set of indices.
+ *
+ * The "split" function divides each input table into multiple intervals
+ * of rows using the indices values and it stores the intervals into the output
+ * tables. Regarding the interval of indices, a pair of values are taken from
+ * the indices array in a consecutive manner. The pair of indices are left-closed
+ * and right-open.
+ *
+ * The indices array ('splits') is require to be a monotonic non-decreasing set.
+ * The indices in the array are required to comply with the following conditions:
+ * a, b belongs to Range[0, number of rows in input table]
+ * a <= b, where the position of a is less or equal to the position of b.
+ *
+ * The split function will take a pair of indices from the indices array
+ * ('splits') in a consecutive manner. For the first pair, the function will
+ * take the value 0 and the first element of the indices array. For the last pair,
+ * the function will take the last element of the indices array and the number
+ * of rows in the table.
+ *
+ * Exceptional cases for the indices array are:
```

Comment:
Likewise update all the other documentation with the `@throws` instead of "Exceptional cases".

Reply:
moved table `split` and `slice` out of this PR.